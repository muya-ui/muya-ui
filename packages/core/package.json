{
  "name": "@muya-ui/core",
  "version": "0.4.21",
  "author": "Muya UI Team",
  "title": "Muya UI Components",
  "description": "Muya UI - The core library of Muya UI",
  "repository": "https://github.com/muya-ui/muya-ui.git",
  "bugs": {
    "url": "https://github.com/muya-ui/muya-ui/issues"
  },
  "files": [
    "dist",
    "lib",
    "es",
    "typings",
    "types"
  ],
  "main": "lib/index.js",
  "module": "es/index.js",
  "typings": "types/index.d.ts",
  "license": "MIT",
  "scripts": {
    "clean": "cross-env rimraf lib && cross-env rimraf es && cross-env rimraf types && cross-env rimraf tsconfig-build.tsbuildinfo",
    "build": "tsc --version && npm run clean && npm run build:types && npm run build:cjs && npm run build:es",
    "build:types": "cross-env tsc -b ./tsconfig-build.json",
    "build:cjs": "cross-env BABEL_ENV=cjs babel --root-mode upward src -d lib --ignore **/*.test.ts,**/*.test.tsx,**/demos/*.ts,**/demos/*.tsx --extensions \".ts,.tsx\"",
    "build:es": "cross-env BABEL_ENV=es babel --root-mode upward src -d es --ignore **/*.test.ts,**/*.test.tsx,**/demos/*.ts,**/demos/*.tsx --extensions \".ts,.tsx\""
  },
  "peerDependencies": {
    "react": ">=16.8.0",
    "react-dom": ">=16.8.0",
    "@types/react": ">=16.9.0",
    "@types/react-dom": ">=16.9.0",
    "styled-components": ">=4.3.2"
  },
  "peerDependenciesMeta": {
    "@types/react": {
      "optional": true
    },
    "@types/react-dom": {
      "optional": true
    }
  },
  "dependencies": {
    "@muya-ui/utils": "0.1.3",
    "@muya-ui/theme-light": "0.3.18",
    "@types/lodash": "^4.14.149",
    "@types/react-transition-group": "^4.2.0",
    "@types/reactour": "^1.13.1",
    "array-tree-filter": "^2.1.0",
    "async-validator": "^3.2.2",
    "dayjs": "^1.8.15",
    "lodash": "^4.17.15",
    "popper.js": "^1.15.0",
    "react-dropzone": "^10.1.8",
    "react-is": "^16.9.0",
    "react-transition-group": "^4.2.1",
    "reactour": "^1.16.0",
    "resize-observer-polyfill": "^1.5.1",
    "shallow-equal": "^1.2.0",
    "warning": "^4.0.3"
  },
  "devDependencies": {
    "@babel/cli": "^7.5.5",
    "cross-env": "^5.2.0",
    "fetch-jsonp": "^1.1.3",
    "querystring": "^0.2.0",
    "rimraf": "^3.0.0",
    "typescript": "3.5.3"
  },
  "sideEffects": false,
  "engines": {
    "node": ">=8.0.0"
  }
}
