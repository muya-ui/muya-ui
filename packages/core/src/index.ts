/* eslint-disable */
/// <reference path="../typings/index.d.ts" />

// 主题相关
export * from './ThemeProvider';
export * from './utils/useTheme';
export * from './utils/withTheme';
export * from './utils/useStyles';
export * from './utils';

// 交互组件
export * from './Popper';
export * from './Portal';
export * from './Swipe';
export * from './Trigger';

// 动画组件
export * from './Animation';

// 基础组件
export * from './Alert';
export * from './AutoComplete';
export * from './Badge';
export * from './Breadcrumbs';
export * from './Background';
export * from './Button';
export * from './Calendar';
export * from './Carousel';
export * from './Cascader';
export * from './Checkbox';
export * from './DatePicker';
export * from './DateTimePicker';
export * from './WeekPicker';
export * from './Dialog';
export * from './Drawer';
export * from './Dropdown';
export * from './EllipsisContent';
export * from './Form';
export * from './Grid';
export * from './Img';
export * from './ImgCropper';
export * from './ImgPreview';
export * from './Input';
export * from './InputNumber';
export * from './IconButton';
export * from './Link';
export * from './Menu';
export * from './Notification';
export * from './Popconfirm';
export * from './PopoverCard';
export * from './Progress';
export * from './Radio';
export * from './Select';
export * from './Slider';
export * from './Skeleton';
export * from './Spin';
export * from './Steps';
export * from './Switch';
export * from './Pagination';
export * from './Result';
export * from './Table';
export * from './Tabs';
export * from './Tag';
export * from './toast';
export * from './Tooltip';
export * from './TimePicker';
export * from './Tree';
export * from './TreeSelect';
export * from './Typography';
export * from './Upload';
export * from './InputGroup';
export * from './Guide';
export * from './Space';
export * from './Collapse';
export * from './ScrollView';
export * from './Sticky';
export * from './Anchor';

// 业务组件
export * from './Card';

// i18n
export * from './Locale';

// 一些通用类型
export { ILabeledValue, IReactElementWithRef } from './types';
