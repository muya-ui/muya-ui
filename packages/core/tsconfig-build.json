{
  "extends": "../../tsconfig.json",
  "include": ["src/**/*"],
  "exclude": [
    "src/**/*.test.ts",
    "src/**/*.test.tsx",
    "**/demos/*.ts",
    "**/demos/*.tsx",
    "typings"
  ],
  "compilerOptions": {
    "rootDir": "./src",
    "outDir": "./types",
    "composite": true,
    "emitDeclarationOnly": true,
    "skipLibCheck": true
  },
  "references": [
    {
      "path": "../utils/tsconfig-build.json"
    },
    {
      "path": "../theme-light/tsconfig-build.json"
    }
  ]
}
