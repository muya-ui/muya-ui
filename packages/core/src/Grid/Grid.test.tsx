import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import { Col, Row } from '@muya-ui/core';

describe('Grid', () => {
  test('Grid Snapshot', () => {
    const offset = renderer
      .create(
        <Row>
          <Col span={8}>col</Col>
          <Col span={8} offset={8}>
            col-8 col-offset-8
          </Col>
        </Row>,
      )
      .toJSON();
    expect(offset).toMatchSnapshot();
    const pushAndPull = renderer
      .create(
        <Row>
          <Col span={18} push={6}>
            col-18 col-push-6
          </Col>
          <Col span={6} pull={18}>
            col-6 col-pull-18
          </Col>
        </Row>,
      )
      .toJSON();
    expect(pushAndPull).toMatchSnapshot();
    const order = renderer
      .create(
        <Row>
          <Col className="col" span={6} order={4}>
            1 col-order-4
          </Col>
          <Col className="col" span={6} order={3}>
            2 col-order-3
          </Col>
          <Col className="col" span={6} order={2}>
            3 col-order-2
          </Col>
          <Col className="col" span={6} order={1}>
            4 col-order-1
          </Col>
        </Row>,
      )
      .toJSON();
    expect(order).toMatchSnapshot();
    const views = Array(20).fill(0);
    const equalNum = renderer
      .create(
        <Row equalNum={{ md: 4, lg: 5 }} gutter={{ xs: 10, lg: 20 }}>
          {views.map((i, index) => (
            <Col key={index} className="col">
              Col-{index}
            </Col>
          ))}
        </Row>,
      )
      .toJSON();
    expect(equalNum).toMatchSnapshot();
  });
  test('Grid Responsive', () => {
    mount(
      <Row gutter={{ xs: 10, lg: 20 }}>
        <Col xs={2} sm={4} md={6} lg={8} xl={10}>
          Col
        </Col>
        <Col xs={20} sm={16} md={12} lg={8} xl={4}>
          Col
        </Col>
        <Col xs={2} sm={4} md={6} lg={8} xl={10}>
          Col
        </Col>
      </Row>,
    );
  });
});
