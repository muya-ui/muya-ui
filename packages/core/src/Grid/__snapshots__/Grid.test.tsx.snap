// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Grid Snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  box-sizing: border-box;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

0 .c0::before,
.c0::after {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  position: relative;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  box-sizing: border-box;
  min-height: 1px;
  width: 33.33333333333333%;
}

.c2 {
  position: relative;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  box-sizing: border-box;
  min-height: 1px;
  width: 33.33333333333333%;
  margin-left: 33.33333333333333%;
}

<div
  className="c0 muya-row-wrapper"
  style={Object {}}
>
  <div
    className="c1 muya-col-wrapper"
    style={Object {}}
  >
    col
  </div>
  <div
    className="c2 muya-col-wrapper"
    style={Object {}}
  >
    col-8 col-offset-8
  </div>
</div>
`;

exports[`Grid Grid Snapshot 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  box-sizing: border-box;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

0 .c0::before,
.c0::after {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  position: relative;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  box-sizing: border-box;
  min-height: 1px;
  width: 75%;
  left: 25%;
}

.c2 {
  position: relative;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  box-sizing: border-box;
  min-height: 1px;
  width: 25%;
  right: 75%;
}

<div
  className="c0 muya-row-wrapper"
  style={Object {}}
>
  <div
    className="c1 muya-col-wrapper"
    style={Object {}}
  >
    col-18 col-push-6
  </div>
  <div
    className="c2 muya-col-wrapper"
    style={Object {}}
  >
    col-6 col-pull-18
  </div>
</div>
`;

exports[`Grid Grid Snapshot 3`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  box-sizing: border-box;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

0 .c0::before,
.c0::after {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  position: relative;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  box-sizing: border-box;
  min-height: 1px;
  width: 25%;
  -webkit-order: 4;
  -ms-flex-order: 4;
  order: 4;
}

.c2 {
  position: relative;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  box-sizing: border-box;
  min-height: 1px;
  width: 25%;
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
}

.c3 {
  position: relative;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  box-sizing: border-box;
  min-height: 1px;
  width: 25%;
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
}

.c4 {
  position: relative;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  box-sizing: border-box;
  min-height: 1px;
  width: 25%;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

<div
  className="c0 muya-row-wrapper"
  style={Object {}}
>
  <div
    className="c1 muya-col-wrapper col"
    style={Object {}}
  >
    1 col-order-4
  </div>
  <div
    className="c2 muya-col-wrapper col"
    style={Object {}}
  >
    2 col-order-3
  </div>
  <div
    className="c3 muya-col-wrapper col"
    style={Object {}}
  >
    3 col-order-2
  </div>
  <div
    className="c4 muya-col-wrapper col"
    style={Object {}}
  >
    4 col-order-1
  </div>
</div>
`;

exports[`Grid Grid Snapshot 4`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  box-sizing: border-box;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-left: -5px;
  margin-right: -5px;
}

.c0::before,
.c0::after {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  position: relative;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  box-sizing: border-box;
  min-height: 1px;
  width: 25%;
  padding-left: 5px;
  padding-right: 5px;
}

<div
  className="c0 muya-row-wrapper"
  style={Object {}}
>
  <div
    className="c1 muya-col-wrapper col"
    style={Object {}}
  >
    Col-
    0
  </div>
  <div
    className="c1 muya-col-wrapper col"
    style={Object {}}
  >
    Col-
    1
  </div>
  <div
    className="c1 muya-col-wrapper col"
    style={Object {}}
  >
    Col-
    2
  </div>
  <div
    className="c1 muya-col-wrapper col"
    style={Object {}}
  >
    Col-
    3
  </div>
  <div
    className="c1 muya-col-wrapper col"
    style={Object {}}
  >
    Col-
    4
  </div>
  <div
    className="c1 muya-col-wrapper col"
    style={Object {}}
  >
    Col-
    5
  </div>
  <div
    className="c1 muya-col-wrapper col"
    style={Object {}}
  >
    Col-
    6
  </div>
  <div
    className="c1 muya-col-wrapper col"
    style={Object {}}
  >
    Col-
    7
  </div>
  <div
    className="c1 muya-col-wrapper col"
    style={Object {}}
  >
    Col-
    8
  </div>
  <div
    className="c1 muya-col-wrapper col"
    style={Object {}}
  >
    Col-
    9
  </div>
  <div
    className="c1 muya-col-wrapper col"
    style={Object {}}
  >
    Col-
    10
  </div>
  <div
    className="c1 muya-col-wrapper col"
    style={Object {}}
  >
    Col-
    11
  </div>
  <div
    className="c1 muya-col-wrapper col"
    style={Object {}}
  >
    Col-
    12
  </div>
  <div
    className="c1 muya-col-wrapper col"
    style={Object {}}
  >
    Col-
    13
  </div>
  <div
    className="c1 muya-col-wrapper col"
    style={Object {}}
  >
    Col-
    14
  </div>
  <div
    className="c1 muya-col-wrapper col"
    style={Object {}}
  >
    Col-
    15
  </div>
  <div
    className="c1 muya-col-wrapper col"
    style={Object {}}
  >
    Col-
    16
  </div>
  <div
    className="c1 muya-col-wrapper col"
    style={Object {}}
  >
    Col-
    17
  </div>
  <div
    className="c1 muya-col-wrapper col"
    style={Object {}}
  >
    Col-
    18
  </div>
  <div
    className="c1 muya-col-wrapper col"
    style={Object {}}
  >
    Col-
    19
  </div>
</div>
`;
